pkg/commands/status.go

```diff
@@ -357,34 +331,194 @@ func printPrefix(p string) string {
+func getWorkloadInstanceStatusAndCreationTime(ctx context.Context, c client.Client, ns string, wlRef runtimev1alpha1.TypedReference) (string, bool, metav1.Time, error) {
 	wlUnstruct := unstructured.Unstructured{}
 	wlUnstruct.SetGroupVersionKind(wlRef.GroupVersionKind())
 	if err := c.Get(ctx, client.ObjectKey{Namespace: ns, Name: wlRef.Name},
 		&wlUnstruct); err != nil {
-		return "", err
+		return "", false, metav1.Time{}, err
 	}
+	ct := wlUnstruct.GetCreationTimestamp()
+
 	statusData, foundStatus, _ := unstructured.NestedMap(wlUnstruct.Object, "status")
 	if foundStatus {
 		statusYaml, err := yaml.Marshal(statusData)
 		if err != nil {
-			return "", err
+			return "", false, ct, err
+		}
+		return string(statusYaml), true, ct, nil
+	}
+	return "", false, ct, nil
+}
+
+func trackInitializeStatus(ctx context.Context, c client.Client, compName, appName string, env *types.EnvMeta) (CompStatus, string, error) {
+	app, appConfig, err := getAppAndApplicationConfiguration(ctx, c, compName, appName, env)
+	if err != nil {
+		return compStatusUnknown, "", err
+	}
+	if app == nil || appConfig == nil {
+		return compStatusUnknown, "", errors.New(ErrNotLoadAppConfig)
+	}
+	_, foundWlStatus := getWorkloadStatusFromAppConfig(appConfig, compName)
+	appConfigReconcileStatus := appConfig.Status.GetCondition(runtimev1alpha1.TypeSynced).Status
+	switch appConfigReconcileStatus {
+	case corev1.ConditionUnknown:
+		return compStatusInitializing, "", nil
+	case corev1.ConditionTrue:
+		if foundWlStatus {
+			return compStatusInitialized, "", nil
+		}
+		return compStatusInitializing, "", nil
+	case corev1.ConditionFalse:
+		appConfigConditionMsg := appConfig.Status.GetCondition(runtimev1alpha1.TypeSynced).Message
+		return compStatusInitFail, appConfigConditionMsg, nil
+	}
+	return compStatusInitializing, "", nil
+}
+
+func trackDeployStatus(ctx context.Context, c client.Client, compName, appName string, env *types.EnvMeta) (CompStatus, string, error) {
+	app, appConfig, err := getAppAndApplicationConfiguration(ctx, c, compName, appName, env)
+	if err != nil {
+		return compStatusUnknown, "", err
+	}
+	if app == nil || appConfig == nil {
+		return compStatusUnknown, "", errors.New(ErrNotLoadAppConfig)
+	}
+
+	wlStatus, foundWlStatus := getWorkloadStatusFromAppConfig(appConfig, compName)
+	// make sure component already initilized
+	if !foundWlStatus {
+		appConfigConditionMsg := appConfig.Status.GetCondition(runtimev1alpha1.TypeSynced).Message
+		return compStatusUnknown, "", fmt.Errorf(ErrFmtNotInitialized, appConfigConditionMsg)
+	}
+	wlRef := wlStatus.Reference
+
+	_, foundStatus, ct, err := getWorkloadInstanceStatusAndCreationTime(ctx, c, env.Namespace, wlRef)
+	if err != nil {
+		return compStatusUnknown, "", err
+	}
+	if foundStatus {
+		//TODO(roywang) temporarily use status to judge workload controller is running
+		// even not every workload has `status` field
+		return compStatusDeployed, "", nil
+	}
+
```

Comment:
we should add a `TODO` here, because an OAM deploy including `workload + trait`, but I can only see workload checked here, if route trait not work, the app also won't be able to access.

Reply:
Fixed.